network/src/transport.rs

```diff
@@ -184,7 +184,11 @@ pub fn build_tcp_noise_transport(
     identity_keypair: (X25519StaticPrivateKey, X25519StaticPublicKey),
     trusted_peers: Arc<RwLock<HashMap<PeerId, NetworkPublicKeys>>>,
 ) -> boxed::BoxedTransport<(Identity, impl StreamMultiplexer), impl ::std::error::Error> {
-    let tcp_transport = tcp::TcpTransport::default();
+    let tcp_transport = tcp::TcpTransport {
+        nodelay: Some(true),
+        ..tcp::TcpTransport::default()
+    };
```

Comment:
I think we just need to add nodelay to `build_permissionless_tcp_noise_transport` and `build_tcp_transport`. Otherwise LGTM

@bors-libra delegate+

Reply:
Done